model/variable.cc

```diff
@@ -415,6 +415,16 @@ bool VariableBase::visible() const
   auto g=group.lock();
   //toplevel i/o items always visible
   if ((!g || !g->group.lock()) && g==controller.lock()) return true;
+  // ensure pars and constants with invisible out wires are made invisible. for ticket 1275  
+  if (type()==constant || type()==parameter)
+  {
+	if (std::any_of(ports[0]->wires().begin(),ports[0]->wires().end(), [](Wire* w){return w->attachedToDefiningVar() && !w->visible();})) return false;
+	else return true;
```

Comment:
Clever, but I think you should be using all_of rather than any_of

Reply:
Hmmm. I knew there was a reason why I used any_of instead of all_of. When I use all_of all parameters disappear from the canvas... Reverting now...